Updating twitter variables

julka opened the pull request:
Updating module to use passthrough twitter via and twitter text variables in addthis_share, as twitter no longer supports the template option (issue 2505439)

Commit:
Updating plugin to use passthrough twitter via and twitter text variables in addthis_share, as twitter no longer supports the template option (issue 2505439)

matglas commented:
Looks good. I do however miss a update method to remove the old variable. And could you add a default for this new variable too in the update and install hooks?

Commit:
deleting twitter variables on uninstall

Commit:
update function for deleting the twitter template variable and adding the twitter text variable

Commit:
changelog updates

julka commented:
I'm happy to. 

I created function `addthis_update_7403` which deletes the old variable and adds the new one with its default value. I added both the twitter text and twitter via variables to the list of ones that should be deleted on uninstall (I'm not sure you wanted this though, since the old twitter variable wasn't being deleted on uninstall).

I also updated the change log to add notes about this change and the change for the default services that you merged yesterday.

Can you point me at an example of where you set the default value of a field on install? I'm not experienced with Drupal. The only thing I found that seems like it might do that is the `addthis_field_schema` function, but that hook doesn't seem to get called by anything in our plugin. There's also the admin form defaults in `includes/addthis.admin.inc`, but the default value for this feild is already getting set there.

matglas commented on CHANGELOG.txt:
```diff
@@ -2,6 +2,8 @@
 
 - Issue #1960656 by juampynr: Give other modules the chance to hook in once addthis variable is available [#1960656]
 - Issue #2407083 by petterw03: Add support for medium sized icons (20x20) [#2407083]
+- Issue #2505439 by julka: update Twitter settings to use supported variables (addthis_config.passthrough.twitter.via and addthis_config.passthrough.twitter.text instead of addthis_config.twmplates.twitter)
```
Here is a typo in the .templates.twitter.

julka replied:
fixed

matglas commented on addthis.install:
```diff
@@ -96,3 +98,12 @@ function addthis_update_7402() {
+ */
+function addthis_update_7403() {
+  variable_del('addthis_twitter_template');
+  variable_set(AddThis::TWITTER_TEXT_KEY, TWITTER_TEXT_DEFAULT);
```
I asked for setting the default but I was wrong. You are already loading a default in the ->getTwitterText(). So thats enough. You can remove this line.

julka replied:
removed

Commit:
fixing typo (twmplates -> templates)

Commit:
removing unnecessary setting of default value

matglas commented on CHANGELOG.txt:
```diff
@@ -2,6 +2,8 @@
 - Issue #1960656 by juampynr: Give other modules the chance to hook in once addthis variable is available [#1960656]
 - Issue #2407083 by petterw03: Add support for medium sized icons (20x20) [#2407083]
+- Issue #2505439 by julka: update Twitter settings to use supported variables (addthis_config.passthrough.twitter.via and addthis_config.passthrough.twitter.text instead of addthis_config.templates.twitter)
+- Modified default addthis_toolbox icons from facebook,twitter to preferred_1,preferred_2,preferred_3,preferred_4,compact as this provides more functionality and the most shares.
```
I added this update already to the CHANGELOG. I added a section with github merges. See here: http://cgit.drupalcode.org/addthis/commit/?id=c2073940a9e6383ee0a31bc378981cafc0999d55

julka replied:
Removed. Sorry, I forgot that github doesn't automatically keep my fork up to date with yours and missed that change.

And please don't hesitate to modify of completely change my change log entry (or ask me to). I was just trying to be thorough. I'm not attached to any of the wording.

julka committed:
Merge pull request #1 from matglas/7.x-4.x

updating my fork

Commit:
Merge remote-tracking branch 'github/7.x-4.x' into issues/2505439-twitter-configs

Commit:
cleanup after updating my fork, and this branch
